oneview_redfish_toolkit/api/event.py

```diff
@@ -20,42 +20,79 @@
     import RedfishJsonValidator
 
 
+ONEVIEW_TO_REDFISH_EVENTS = {
+    "Created": "ResourceAdded",
+    "Updated": "ResourceUpdated",
+    "Deleted": "ResourceRemoved",
+
+    # OneView does not have an equivalent change type to StatusChange
+    "StatusChange": "StatusChange"
+}
+
+
 class Event(RedfishJsonValidator):
     """Creates a Event Redfish dict
 
         Populates self.redfish with some hardcoded Event
-        values and with the response from OneView.
+        values and with the SCMB message received from OneView.
     """
 
     SCHEMA_NAME = 'Event'
 
-    def __init__(self, oneview_alert, event_type):
+    def __init__(self, oneview_message):
         """Event constructor
 
-            Populates self.redfish with some hardcoded Event
-            values and with the response from OneView.
+        Populates self.redfish with some hardcoded Event
+        values and with the SCMB message received from OneView.
 
             Args:
-                oneview_alert: A dict with an OneView alert message
-                event_type: string with redfish event_type
+                oneview_message: A dict with an OneView SCMB message
         """
 
         super().__init__(self.SCHEMA_NAME)
 
-        id = str(os.path.basename(oneview_alert["resourceUri"]))
+        category = oneview_message["resource"]["category"]
 
         self.redfish["@odata.type"] = "#Event.v1_2_0.Event"
-        self.redfish["Id"] = id
-        self.redfish["Name"] = oneview_alert["resource"]["description"]
-        self.redfish["Events"] = []
         self.redfish["@odata.context"] = "/redfish/v1/$metadata#Event.Event"
-        event_id = 0
-        event = {}
-        event["EventId"] = str(event_id)
-        event["EventType"] = event_type
-        # Nothing good matches here!!!!!
-        event["MessageId"] = "Base.1.1.Success"
-        event["Severity"] = oneview_alert["resource"]["severity"]
-        event["EventTimestamp"] = oneview_alert["resource"]["created"]
-        self.redfish["Events"].append(event)
+        self.redfish["Events@odata.count"] = 1
+        self.redfish["Events"] = list()
+
+        if (category == "alerts"):
+            self.build_event_from_alert(oneview_message)
+        else:
+            self.build_event_from_task(oneview_message)
```

Comment:
Is the message here actually just a OneView resource and not necessarily a OneView Task?

Reply:
At this point, we have an SCMB message, which can be an alert or a task. According to the OneView documentation, the JSON payload of each message from the SCMB contains a `resource` attribute, from which we can get the `category` field value and check whether it is a task or an alert message.